src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,170 @@
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+
+public class LocalFile implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+    //Instance varibles for saving the current data of server.
+    private final HashMap<Uuid,User> users;
+    private final HashMap<Uuid,ConversationHeader> conversationHeaders;
+    private final HashMap<Uuid,Message> messages;
+
+    private transient final File file;
+
+    private static final Logger.Log LOG = Logger.newLog(LocalFile.class);
+
+    private transient boolean hasModified = false;//It indicates if there is a new data should be handled.
+    public LocalFile(File file)
+    {
+        this.file = file;
+        LocalFile newClass;
+        HashMap<Uuid,User>tempUsers = new HashMap<>();
+        HashMap<Uuid,ConversationHeader>tempConversationHeaders= new HashMap<>();
+        HashMap<Uuid,Message>tempMessages= new HashMap<>();
+        //Deserialization
+        try(FileInputStream fileInputStream = new FileInputStream(file);
+		    ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream))
+        {
+            //Read the information to current instance
+		    newClass = (LocalFile)objectInputStream.readObject();
+            tempUsers = newClass.users;
+            tempConversationHeaders = newClass.conversationHeaders;
+            tempMessages = newClass.messages;
+        }
+        catch(IOException exception)
+        {
+            if(!file.exists())
+            {
+                System.out.println("WARNING:File doesn't exist, a new local file will be created.");
+            }
+            else
+            {
+                exception.printStackTrace();
+                System.out.println("ERROR:Unable to read data.");
+                throw new RuntimeException("ERROR:Unable to run the program. Local file cannot be read.");
+            }
+        }
+        catch(ClassNotFoundException exception)
+        {
+            System.out.println("ERROR:Failed to desrialized the local data.");
+            exception.printStackTrace();
+            throw new RuntimeException("ERROR:Unable to run the program. Local file cannot be read.");
+        }
+       //Write temporary information to the instance varibles
+       users = tempUsers;
+       conversationHeaders = tempConversationHeaders;
+       messages = tempMessages;
+    }
+    /**
+     * Save the data into local file.
+     * 
+     * @exception   Failed to save data.
+     */
+    public void saveData() throws IOException
+    {
+        if(!hasModified)//Only save the data when the data is updated.
+        {
+            return;
+        }
+
+        if(!file.exists())
+        {
+            file.createNewFile();
+        }
+        //Serialization
+        try(FileOutputStream fileOutputStream = new FileOutputStream(file);
+        ObjectOutputStream objectOutputStream = new ObjectOutputStream(fileOutputStream))
+        {
+         objectOutputStream.writeObject(this); 
```

Comment:
See my above comment - rather than using java object serialization, I'd recommend using the custom SERIALIZER classes in the message, conversation, and user objects instead.